### PR TITLE
libgit2: version update to @1.4.2

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Version update, nothing more here.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6 PPC (10A190)
Xcode 3.2

macOS 10.6.8 Server – ppc and x86_64
Xcode 3.2.6

### CODE REVIEW

**Comment by herbygillot** [comment]
Upgrading to 1.4.2 is going to break a lot of software that still expects the 1.3 branch.  Safest thing to do here is the latest 1.3, which is 1.3.1

Not sure what the best way forward is to also have the 1.4 branch as well.

File: devel/libgit2/Portfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ PortGroup           legacysupport 1.0
 PortGroup           muniversal 1.0
 
 # don't forget to update py-pygit2 and libgit2-glib as well
-github.setup        libgit2 libgit2 1.3.0 v
+github.setup        libgit2 libgit2 1.4.2 v
 github.tarball_from archive
 revision            1
```
Comment (reneeotten) [inline devel/libgit2/Portfile:12]: the revision field should be reset to zero when updating the version

Comment (barracuda156) [reply devel/libgit2/Portfile:12]: @reneeotten Thank you. Fixed.

**Review by mascguy — APPROVED** [review]

**Comment by herbygillot** [comment]
> Upgrading to 1.4.2 is going to break a lot of software that still expects the 1.3 branch. Safest thing to do here is the latest 1.3, which is 1.3.1
> 
> Not sure what the best way forward is to also have the 1.4 branch as well.

Besides the above, this change should also address `py-pygit2` and `libgit2-glib` as mentioned in the top-level comment in the Portfile.

**Comment by barracuda156** [comment]
> > Upgrading to 1.4.2 is going to break a lot of software that still expects the 1.3 branch. Safest thing to do here is the latest 1.3, which is 1.3.1
> > Not sure what the best way forward is to also have the 1.4 branch as well.
> 
> Besides the above, this change should also address `py-pygit2` and `libgit2-glib` as mentioned in the top-level comment in the Portfile.

@herbygillot `libgit2-glib` update: https://github.com/macports/macports-ports/pull/14618

**Comment by herbygillot** [comment]
> > > Upgrading to 1.4.2 is going to break a lot of software that still expects the 1.3 branch. Safest thing to do here is the latest 1.3, which is 1.3.1
> > > Not sure what the best way forward is to also have the 1.4 branch as well.
> > 
> > 
> > Besides the above, this change should also address `py-pygit2` and `libgit2-glib` as mentioned in the top-level comment in the Portfile.
> 
> @herbygillot `libgit2-glib` update: #14618 And `py-pygit2`: #14619

It probably makes sense to stack the commits for `libgit2-glib` and `py-pygit2` within this PR after the `libgit2` commit to verify that they build correctly.

Also my point about libgit2 version `1.4.2`, have you tested that other downstream ports that have `libgit2` as a dependency can build against `1.4.2`?

**Comment by barracuda156** [comment]
@herbygillot Unfortunately, neither rust no go build for PPC atm, so I am unable to check most of dependencies, someone has to do that. I can try gnuastro, if required dependencies build.

**Comment by barracuda156** [comment]
`gnuastro` has built with `libgit2` @1.4.2:
```
36-248% port -v installed gnuastro
The following ports are currently installed:
  gnuastro @0.17_0 (active) requested_variants='' platform='darwin 10' archs='ppc' date='2022-04-19T08:21:31+0800'
```

**Comment by herbygillot** [comment]
`turbogit` fails to build on `libgit2` 1.4:

```
make: Entering directory `/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_devel_turbogit/turbogit/work/turbogit-3.1.1'
go build -trimpath -o "bin/tug" -ldflags='-s -w -X "github.com/b4nst/turbogit/cmd.BuildDate=Mon May 02 06:52:28 2022" -X "github.com/b4nst/turbogit/cmd.Commit=refs/tags/v3.1.1" -X "github.com/b4nst/turbogit/cmd.Version=3.1.1"' ./main.go
# github.com/libgit2/git2go/v33
../.home/go/pkg/mod/github.com/libgit2/git2go/v33@v33.0.4/Build_system_dynamic.go:12:3: error: "Invalid libgit2 version; this git2go supports libgit2 between v1.3.0 and v1.3.0"
# error "Invalid libgit2 version; this git2go supports libgit2 between v1.3.0 and v1.3.0"
  ^
1 error generated.
make: *** [bin/tug] Error 2
make: Leaving directory `/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_devel_turbogit/turbogit/work/turbogit-3.1.1'
Command failed:  cd "/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_devel_turbogit/turbogit/work/turbogit-3.1.1" && /usr/bin/make -j10 -w build
Exit code: 2
```

**Comment by barracuda156** [comment]
> `turbogit` fails to build on `libgit2` 1.4:
> 
> ```
> make: Entering directory `/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_devel_turbogit/turbogit/work/turbogit-3.1.1'
> go build -trimpath -o "bin/tug" -ldflags='-s -w -X "github.com/b4nst/turbogit/cmd.BuildDate=Mon May 02 06:52:28 2022" -X "github.com/b4nst/turbogit/cmd.Commit=refs/tags/v3.1.1" -X "github.com/b4nst/turbogit/cmd.Version=3.1.1"' ./main.go
> # github.com/libgit2/git2go/v33
> ../.home/go/pkg/mod/github.com/libgit2/git2go/v33@v33.0.4/Build_system_dynamic.go:12:3: error: "Invalid libgit2 version; this git2go supports libgit2 between v1.3.0 and v1.3.0"
> # error "Invalid libgit2 version; this git2go supports libgit2 between v1.3.0 and v1.3.0"
>   ^
> 1 error generated.
> make: *** [bin/tug] Error 2
> make: Leaving directory `/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_devel_turbogit/turbogit/work/turbogit-3.1.1'
> Command failed:  cd "/opt/macports-test/var/macports/build/_Users_herby_Source_macports-ports_devel_turbogit/turbogit/work/turbogit-3.1.1" && /usr/bin/make -j10 -w build
> Exit code: 2
> ```

@herbygillot Yes, I can confirm it fails. I don’t understand their build system, so opened a ticket with upstream. Let’s see if they will be kind enough to fix this failure.

**Comment by b4nst** [comment]
Hey, thanks for trying to update it. I'm still waiting for https://github.com/libgit2/git2go to release a v34 or to fix https://github.com/libgit2/git2go/issues/899 so I can bump the lib (or fix the build).

I can also give a try to the main branch, but I would like to avoid depending on a non-released commit.

**Comment by b4nst** [comment]
Some (late) news. I finally decided to statically link libgit2 into `turbogit`. I'm gonna update the Port so it stops blocking this PR.

**Comment by barracuda156** [comment]
> Some (late) news. I finally decided to statically link libgit2 into `turbogit`. I'm gonna update the Port so it stops blocking this PR.

Great, thanks.

**Comment by judaew** [comment]
@dbevans just updated libgit2 to 1.5.0. See https://github.com/macports/macports-ports/commit/7e482c2ec7a28dd04b81fd0a2fe8395492a1446e

**Pull request closed without merging**